### ISSUE TITLE
Feature request: option to dry-run

### ISSUE THREAD

**Issue by yermulnik** [warrensbox/terraform-switcher#154]
I'd like to ask to implement a feature which would allow me to run `tfswitch` in a dry-run mode where it only outputs whether it would switch terraform version in current directory or not with different exit codes for the two operations (e.g. `0` for when no switching is required and `1` for when `tfswitch` would switch versions).

My use case: we're managing different infrastructures from different repositories including legacy setups – most of these are using different versions of terraform and we're not planning to unify them. Most of the time I need to switch terraform version when I work on something in several different setups at the same time but it's not always that I need to actually switch terraform version (e.g. when I only need to cd into dir to check something) hence suggested `cdtfswitch()` Bash alias is not something I indeed need but would rather want to inject something like `tfswitch --dry-run >/dev/null || echo "tfswitch to use correct terraform version"` into my `PROMPT_COMMAND` env var so that I get notified when my Terraform version doesn't match Terraform version constraint every time I hit enter on the command line.

Thanks in advance.

**Comment by yermulnik** [comment]
As an interim solution (sort of FWIW) here's the ugly bit of Go code I came up with:
```Go
package main

import (
	"fmt"
	"os"

	"github.com/hashicorp/go-version"
	"github.com/jessevdk/go-flags"
)

var opts struct {
	Verbose []bool `short:"v" long:"verbose" description:"Show verbose debug information"`
}

func usage() {
	fmt.Fprintf(os.Stderr, "Usage: %s [-v] <tf version> <tf version constraint>\n", os.Args[0])
	os.Exit(2)
}

func main() {
	args, err := flags.Parse(&opts)
	if err != nil {
		usage()
	}

	if len(args) != 2 {
		usage()
	}

	v1, err := version.NewVersion(args[0])
	if err != nil {
		fmt.Println("ERROR:", err)
		usage()
	}

	constraints, err := version.NewConstraint(args[1])
	if err != nil {
		fmt.Println("ERROR:", err)
		usage()
	}

	if constraints.Check(v1) {
		if len(opts.Verbose) > 0 {
			fmt.Printf("\"%s\" satisfies constraint \"%s\"\n", v1, constraints)
		}
		os.Exit(0)
	} else {
		if len(opts.Verbose) > 0 {
			fmt.Printf("\"%s\" DOES NOT satisfy constraint \"%s\"\n", v1, constraints)
		}
		os.Exit(1)
	}
}
```
```Bash
> terraform version | head -1
Terraform v0.13.6

> fgrep required_version *.tf
  required_version = "~> 1.0.0"

> tf_version_check -v "$(terraform version -json | jq -r .terraform_version)" "$(awk '/^[[:space:]]+required_version[[:space:]]+/ {print gensub("^.+= \"(.+)\"^M?$","\\1","g");}' *.tf 2>/dev/null)"
"0.13.6" DOES NOT satisfy constraint "~> 1.0.0"

> tf_version_check "$(terraform version -json | jq -r .terraform_version)" "$(awk '/^[[:space:]]+required_version[[:space:]]+/ {print gensub("^.+= \"(.+)\"^M?$","\\1","g");}' *.tf 2>/dev/null)" && echo GOOD || echo BAD
BAD

> tfswitch
Reading configuration from home directory for .tfswitch.toml
Reading required version from terraform file
Reading required version from constraint: ~> 1.0.0
Matched version: 1.0.11
Switched terraform to version "1.0.11"

> tf_version_check -v "$(terraform version -json | jq -r .terraform_version)" "$(awk '/^[[:space:]]+required_version[[:space:]]+/ {print gensub("^.+= \"(.+)\"^M?$","\\1","g");}' *.tf 2>/dev/null)"
"1.0.11" satisfies constraint "~> 1.0.0"

> tf_version_check "$(terraform version -json | jq -r .terraform_version)" "$(awk '/^[[:space:]]+required_version[[:space:]]+/ {print gensub("^.+= \"(.+)\"^M?$","\\1","g");}' *.tf 2>/dev/null)" && echo GOOD || echo BAD
GOOD
```

**Comment by yermulnik** [comment]
Just for information ℹ️ 

The `show-required` feature, related to `dry-run` flag, has been released under [v1.7.0](https://github.com/warrensbox/terraform-switcher/releases/tag/v1.7.0): as opposite to `dry-run`, the `show-required` flag is specifically to show the version required by version constraint(s) or the latest version otherwise:
```
 -R, --show-required
                    Show required (or explicitly requested) version. Defaults to
                    latest version if no constraints found
```
So, in a nutshell, the `dry-run` is to show what `tfswitch` would do (without actually downloading anything), whilst `show-required` is to output required version.

> As an interim solution (sort of FWIW) here's the ugly bit of Go code I came up with:

A more comprehensive feature for this has been released under [v1.8.0](https://github.com/warrensbox/terraform-switcher/releases/tag/v1.8.0):
```
 -n, --match-version-requirement=value
                    Check if the requested version matches the requirement
                    mandated by the configuration (env var, module version
                    constraint, config files). Exit successfully if it does (or
                    if there's no requirement found), otherwise exit with a code
                    of `2` (code of `1` denotes a general error)
```
My expectation for this feature usage would be to accompany it with `--log-level=OFF` flag to suppress any logging output and to let consumer only rely upon exit code.
